commons/ava_testsuite/staking_gecko_network_tests.go

```diff
@@ -123,73 +122,58 @@ func (test StakingNetworkRpcWorkflowTest) GetTimeout() time.Duration {
 	highLevelExtraStakerClient := ava_networks.NewHighLevelGeckoClient(
-		extraStakerClient,
+		nonBootValidatorClient,
 		STAKER_USERNAME,
 		STAKER_PASSWORD)
-	err = highLevelExtraStakerClient.GetFundsAndStartValidating(SEED_AMOUNT, STAKE_AMOUNT)
-	if err != nil {
+	if err := highLevelExtraStakerClient.GetFundsAndStartValidating(SEED_AMOUNT, STAKE_AMOUNT); err != nil {
 		context.Fatal(stacktrace.Propagate(err, "Failed to add extra staker."))
 	}
 
-	// collect set of IDs in network
-	nodeIdSet := map[string]bool{}
-	for serviceId, _ := range allServiceIds {
-		client, err := castedNetwork.GetGeckoClient(serviceId)
-		if err != nil {
-			context.Fatal(stacktrace.Propagate(err, "Could not get client for service with ID %v", serviceId))
-		}
-		id, err := client.InfoApi().GetNodeId()
-		if err != nil {
-			context.Fatal(stacktrace.Propagate(err, "Could not get node ID of service with ID %v", serviceId))
-		}
-		nodeIdSet[id] = true
-	}
+	// Give time for the new validator to propagate via gossip
+	time.Sleep(70 * time.Second)
 
-	logrus.Debugf("Network ID Set: %+v", nodeIdSet)
+	stakerIds[nonBootValidatorServiceId] = true
 
-	// verify bootstrapper peer lists have full set of IDs in network.
-	for serviceId, _ := range allServiceIds {
-		client, err := castedNetwork.GetGeckoClient(serviceId)
-		if err != nil {
-			context.Fatal(stacktrace.Propagate(err, "Could not get client for service with ID %v", serviceId))
-		}
-		peers, err := client.InfoApi().GetPeers()
-		nodeId, err := client.InfoApi().GetNodeId()
-		if err != nil {
-			context.Fatal(stacktrace.Propagate(err, "Could not get peers of service with ID %v", serviceId))
-		}
-		logrus.Debugf("Length of Peer set for node %v: %v", serviceId, len(peers))
-
-		peerSet := map[string]bool{}
-		for _, peer := range peers {
-			peerSet[peer.Id] = true
-		}
-		for expectedNodeId, _ := range nodeIdSet {
-			// Nodes are expected to have all other nodes in the network on their peer list.
-			if nodeId != expectedNodeId {
-				context.AssertTrue(peerSet[expectedNodeId], stacktrace.NewError("Didn't find node ID %v in peer set", expectedNodeId))
-			}
-		}
+	/*
+	After gossip, we expect the peers list to look like:
+	1) No node has itself in its peers list
+	2) The validators will have ALL other nodes in the network (propagated via gossip)
+	3) The non-validators will have all the validators in the network (propagated via gossip)
+	 */
+	if err := verifyNetworkFullyConnected(allServiceIds, stakerIds, allNodeIds, allGeckoClients); err != nil {
+		context.Fatal(stacktrace.Propagate(err, "An error occurred verifying that the network is fully connected after gossip"))
 	}
 }
 
 func (test StakingNetworkFullyConnectedTest) GetNetworkLoader() (testsuite.TestNetworkLoader, error) {
 	return getStakingNetworkLoader(map[int]int{
-		NODE_SERVICE_ID:           NORMAL_NODE_CONFIG_ID,
+		// TODO Once this is split into its own file, make these constants
+		0: NORMAL_NODE_CONFIG_ID,
```

Comment:
@mieubrisse these numbers (0, 1) are recorded twice here. once as a variable scoped to the Run function, and once here.

previously we had resolved this by having them as module-wide constants. in my PR on byzantine behavior I see why thats not ideal. however this strategy has this duplication. is there another option available to us to scope these values across the functions here? can we store them as values in the StakingNetworkFullyConnectedTest struct, for example?

Reply:
based on slack conversation we're going to handle this in another PR